internal/testing/setup/setup.go

```diff
@@ -155,13 +155,14 @@ func NewGCPgRPCConn(ctx context.Context, t *testing.T, endPoint, api string) (*g
 }
 
 // NewAzureTestPipeline creates a new connection for testing against Azure Blob.
-func NewAzureTestPipeline(ctx context.Context, t *testing.T, accountName string, accountKey string) (pipeline pipeline.Pipeline, done func(), httpClient *http.Client) {
+func NewAzureTestPipeline(ctx context.Context, t *testing.T, api, accountName, accountKey string) (pipeline pipeline.Pipeline, done func(), httpClient *http.Client) {
 	mode := recorder.ModeReplaying
 	if *Record {
 		mode = recorder.ModeRecording
 	}
 
 	azMatchers := &replay.ProviderMatcher{
+		Headers: []string{"User-Agent"},
```

Comment:
This ensures that the User-Agent header matches during replay.

Reply:
Ah, I have to comment this out because Azure adds something with the "go" version in it to User-Agent, which varies.